.github/workflows/python-lint.yml

```diff
@@ -20,65 +16,69 @@ jobs:
   black:
     runs-on: ubuntu-latest
     steps:
-      - name: Checkout Code
-        uses: actions/checkout@v1
-      - name: Setup Python
-        uses: actions/setup-python@v2
+      - uses: actions/checkout@v1
+
+      - uses: actions/setup-python@v2
         with:
           python-version: '3.7'
 
       # Cache pip dependencies
       # See also: https://github.com/actions/cache/blob/main/examples.md#python---pip
       - name: Get pip cache dir
-        id: pip-cache
+        id: pip-dir
         run: |
           echo "::set-output name=dir::$(pip cache dir)"
 
-      - name: pip cache
-        uses: actions/cache@v2
+      - uses: actions/cache@v2
+        id: pip-cache
         with:
-          path: ${{ steps.pip-cache.outputs.dir }}
-          key: cactbot-pip-${{ hashFiles('**/requirements.txt') }}
+          path: ${{ steps.pip-dir.outputs.dir }}
+          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
           restore-keys: |
-            cactbot-pip-
+            ${{ runner.os }}-pip-
 
       - name: Install Dependencies
+        if: steps.pip-cache.outputs.cache-hit != 'true'
```

Comment:
There should be not a if-checking like this , I think.
According to [official doc](https://pip.pypa.io/en/stable/reference/pip_install/#caching), the pip cache just decrease the http request times, not for installation.
So the installing step should always run in every time.

Reply:
Ok, that explains the behaviour then.